app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/Dataset.kt

```diff
@@ -0,0 +1,16 @@
+package com.github.HumanLearning2021.HumanLearningApp.model
+
+import java.io.Serializable
+
+interface Dataset: Serializable {
+    val name: String
+    val categories: MutableSet<Category>
```

Comment:
[MutableSet is a subclass of Set](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-mutable-set/#kotlin.collections.MutableSet), you don't need to change it in the interface. please revert this.